reVX/plexos/plexos_plants.py

```diff
@@ -0,0 +1,1006 @@
+
+    def _fill_plants(self, plants):
+        """
+        Fill plants with capacity from supply curve points
+
+        Parameters
+        ----------
+        dist_percentile : int, optional
+            Percentile to use to compute distance threshold using sc_gid to
+            SubStation distance , by default 90
+        lcoe_col : str, optional
+            LCOE column to sort by, by default 'total_lcoe'
+        lcoe_thresh : float, optional
+            LCOE threshold multiplier, exclude sc_gids above threshold,
+            by default 1.3
+        max_workers : int, optional
+            Number of workers to use for plant sc extraction, 1 == serial,
+            > 1 == parallel, None == parallel using all available cpus,
+            by default None
+
+        Returns
+        -------
+        plants : list
+            List of plant supply curve tables
+        """
+        i = 0
+        total_cap = np.sum(self.plant_capacity)
+        while np.any(self.plant_capacity > 0):
+            i_cap = np.sum(self.plant_capacity[self.plant_capacity > 0])
+            logger.info('Allocating sc_gids to plants round {}'
+                        .format(i))
+            sc_gids, dists, bus_dists = self._get_sc_gids(plants, i)
+            self._allocate_sc_gids(sc_gids, dists, bus_dists)
+            cap = np.sum(self.plant_capacity[self.plant_capacity > 0])
+            logger.info('{:.1f} MW allocated in round {}'
+                        .format(i_cap - cap, i))
+            i += 1
+            logger.info('{:.1f} MW allocated out of {:.1f} MW'
+                        .format(total_cap - cap, total_cap))
+            logger.info('{} of {} plants have been filled'
+                        .format(np.sum(self.plant_capacity <= 0), len(self)))
+
+    def fill_plants(self, max_workers=None, points_per_worker=400,
+                    dist_percentile=90, lcoe_col='total_lcoe', lcoe_thresh=1.3,
+                    plants_per_worker=40):
+        """
+        Fill plants with capacity from supply curve points
+
+        Parameters
+        ----------
+        max_workers : int, optional
+            Number of workers to use for point and plant creation, 1 == serial,
+            > 1 == parallel, None == parallel using all available cpus,
+            by default None
+        points_per_worker : int, optional
+            Number of points to create on each worker, by default 400
+        dist_percentile : int, optional
+            Percentile to use to compute distance threshold using sc_gid to
+            SubStation distance , by default 90
+        lcoe_col : str, optional
+            LCOE column to sort by, by default 'total_lcoe'
+        lcoe_thresh : float, optional
+            LCOE threshold multiplier, exclude sc_gids above threshold,
+            by default 1.3
+        plants_per_worker : int, optional
+            Number of plants to identify on each worker, by default 40
+
+        Returns
+        -------
+        plants : list
```

Comment:
Is this actually supposed to return something? Doesn't look like it does

Reply:
nope old docstring